src/foam/core/CompoundContextAgency.js

```diff
@@ -40,6 +40,7 @@ for ( Runnable agent : getAgents() ) {
   } catch (Throwable t) {
     logger.error(t.getMessage(), t);
     e.add(t);
+    throw t;
```

Comment:
@hchoura the throwable is added to the compoundexception. 
You can't throw it.  This is nonsense.

Reply:
Yes, this doesn't make sense. If there is an exception, then we'll rethrow it on line 45 below:
e.maybeThrow();